ast/term_test.go

```diff
@@ -869,6 +873,94 @@ func TestSetCopy(t *testing.T) {
 	}
 }
 
+// Constructs a set, and then has several reader goroutines attempt to
+// concurrently iterate across it. This should pretty consistently
+// hit a race condition around sorting the underlying key slice if
+// the sorting isn't guarded properly.
+func TestSetConcurrentReads(t *testing.T) {
```

Comment:
❓ Does the test fail with race detection enabled and without the fix? Or is it create in such a way that it'll fail its assertion _because_ it's hitting a race condition, regardless of the race detector..? I couldn't tell from reading the comment 🤔

Reply:
Ah, it is designed to fail *because* of the race-condition, even without the race-detector enabled.

It should *also* fail in race-detector mode when the fix isn't present (because it's triggering the same race condition as the evaluator / e2e concurrency tests), although I haven't tested this.